test_data/vw_transaction_data.txt

```diff
@@ -1,2 +1,2 @@
-doc1 hello world:2 |user^action mel-lain^click:3 mel-lain^show
-doc2 |author mel-lain |@default_class^action^qualifier hello^show^twice:5 world^click^first |@default_class world
+doc1 hello world:2 ||trans1 |user mel-lain:3 |action click:3 || |user mel-lain |action show ||
```

Comment:
Here I don't follow new VW syntax.
* ``||trans1`` indicate that what follows belongs to transaction type ``trans1``  (this identifier can be used to set weight and/or skip certain transaction types from the model). 
* ``||`` indicate start of new transaction. Does it belong to the previously defined transaction type? (e.i. in case of doc1 ``trans1``). 
* Why use ``|`` and not ``@`` in ``|user`` ``|action``? To me these just define class ids.

Reply:
Sorry, it's getting late on a clock... I forgot VW syntax. Surely, we use | to indicate class id.

Reply:
But few more picky questions
- what if first || doesn't indicate transaction type? 
- what if user doesn't close last transaction with trailing ``||`` ?
- does |class scope apply to the next transaction? E.i., in `` |class1 ||trans1 token `` does ``|class1`` apply to token? 

To avoid confusion I suggest to use separate syntax to
- indicate start and/or end of transactions
- indicate scope which defines transaction types

Maybe a good option is to put each transaction like this:
```
doc1 hello world:2 <trans1>|user mel-lain:3 |action click:3</trans1><trans1>|user mel-lain |action show</trans1>
```
or shortcut to 
```
doc1 hello world:2 <trans1 |user mel-lain:3 |action click:3><trans1 |user mel-lain |action show>
```
Perhaps ok to require ``trans1`` in each transaction?